set.mm

```diff
@@ -735799,6 +735836,340 @@ their group (addition) operations are equal for all pairs of elements of
+                         <. ( +g ` ndx ) , + >. } $.
+    $( The set of nonnegative integers under (complex) addition is a monoid.
+       Example in [Lang] p. 6.  Remark: ` M ` could have also been written as
+       ` ( CCfld |``s NN0 ) ` .  (Contributed by AV, 27-Dec-2023.) $)
+    nn0mnd $p |- M e. Mnd $=
+      ( vx vy vz ve wcel cv caddc co cn0 wceq wral wa nn0cn jca cc0 eqeq1d cvv
+      cc cmnd wrex nn0addcl w3a 3anim123i 3expa addass syl ralrimiva rgen2 c0ex
+      wex eleq1 oveq1 oveq2 anbi12d ralbidv 0nn0 addid2d addid1d rgen ceqsexv2d
+      pm3.2i df-rex mpbir cbs nn0ex grpbase ax-mp cplusg addex grpplusg ismnd
+      cfv ) AUAGCHZDHZIJZKGZVQEHZIJVOVPVSIJIJLZEKMZNZDKMCKMZFHZVOIJZVOLZVOWDIJZ
+      VOLZNZCKMZFKUBZNWCWKWBCDKKVOKGZVPKGZNZVRWAVOVPUCWNVTEKWNVSKGZNVOTGZVPTGZV
+      STGZUDZVTWLWMWOWSWLWPWMWQWOWRVOOZVPOVSOUEUFVOVPVSUGUHUIPUJWKWDKGZWJNZFULX
+      BQKGZQVOIJZVOLZVOQIJZVOLZNZCKMZNFQUKWDQLZXAXCWJXIWDQKUMXJWIXHCKXJWFXEWHXG
+      XJWEXDVOWDQVOIUNRXJWGXFVOWDQVOIUORUPUQUPXCXIURXHCKWLXEXGWLVOWTUSWLVOWTUTP
+      VAVCVBWJFKVDVEVCKIFACDEKSGKAVFVNLVGKIASBVHVIISGIAVJVNLVKKIASBVLVIVMVE $.
+  $}
+
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Group sum operation (extension 1)
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  ${
+    $d F x $.  $d M x $.  $d N x $.  $d ph x $.
+    gsumsplit1r.b $e |- B = ( Base ` G ) $.
+    gsumsplit1r.p $e |- .+ = ( +g ` G ) $.
+    gsumsplit1r.g $e |- ( ph -> G e. V ) $.
+    gsumsplit1r.m $e |- ( ph -> M e. ZZ ) $.
+    gsumsplit1r.n $e |- ( ph -> N e. ( ZZ>= ` M ) ) $.
+    gsumsplit1r.f $e |- ( ph -> F : ( M ... ( N + 1 ) ) --> B ) $.
+    $( Splitting off the rightmost summand of a group sum.  This corresponds to
+       the (inductive) definition of a (finite) product in [Lang] p. 4, first
+       formula.  (Contributed by AV, 26-Dec-2023.) $)
+    gsumsplit1r $p |- ( ph -> ( G gsum F ) = ( ( G gsum ( F |` ( M ... N ) ) )
+                                             .+ ( F ` ( N + 1 ) ) ) ) $=
+      ( vx co cfv cfz wcel syl cgsu caddc cseq cres cuz peano2uz gsumval2 seqp1
+      c1 wceq wss fzssp1 a1i fssresd uzidd cz eluzfz1 fvresd eqtrd cv wa fzp1ss
+      seq1 sselda seqfveq2 eqtr2d oveq1d 3eqtrd ) AEDUAPGUIUBPZCDFUCZQZGVJQZVID
+      QZCPZEDFGRPZUDZUAPZVMCPABCDEFVIHIJKAGFUEQZSZVIVRSMFGUFTNUGAVSVKVNUJMCDFGU
+      HTAVLVQVMCAVQGCVPFUCZQVLABCVPEFGHIJKMAFVIRPZBVODNVOWAUKAFGULUMUNUGACOVPDF
+      FGAFLUOAFVTQZFVPQZFDQAFUPSZWBWCUJLCVPFVCTAFVODAVSFVOSMFGUQTURUSMAOUTZFUIU
+      BPGRPZSVAWEVODAWFVOWEAWDWFVOUKLFGVBTVDURVEVFVGVH $.
+  $}
+
+  ${
+    $d x y z B $.  $d x y z G $.  $d x y z .+ $.  $d x y z W $.  $d x y z X $.
+    gsumsgrpccat.b $e |- B = ( Base ` G ) $.
+    gsumsgrpccat.p $e |- .+ = ( +g ` G ) $.
+    $( Homomorphic property of not empty composites of a group sum over a
+       semigroup.  Formerly part of proof for ~ gsumccat .  (Contributed by AV,
```

Comment:
It's more of an associative property, so:
"Associative property of the semigroup law: the composition of the iterated compositions of two nonempty finite sequences is equal to the iterated composition of their concatenation. (Recall that "words" are finite sequences, that is, functions on integer intervals ` ( 0 .. N ) ` for some ` N e. NN0 ` .)"
(I'm quoting the last parenthesis from memory, please check.)

Reply:
I would like not to chane the comment until the wording for `gsum`is clarified. The term "Homomorphic property" is taken from the original theorem ~gsumccat. 

And yes, that the sequences are finite is essential here.

Reply:
> (Recall that "words" are finite sequences, that is, functions on integer intervals `( 0 .. N )` for some `N e. NN0` .)" (I'm quoting the last parenthesis from memory, please check.)

No, words are functions on half-open integer intervals `( 0 ..^ N )`, so that empty words are possible. However, empty words must be excluded here (because we cannot assume the existence of an identity element).

Reply:
Ok since it is in your mathbox... But if you want to keep "homomorphic", I'm going to ask you which morphism you're talking about ;) I admit that you can say that the composition law is a morphism when it is associative, but this is a bit contrived.